table/tables/index.go

```diff
@@ -97,6 +97,8 @@ type index struct {
 	tblInfo *model.TableInfo
 	idxInfo *model.IndexInfo
 	prefix  kv.Key
+
+	key []byte // It's used reduce the number of new slice when multiple index keys are created.
```

Comment:
This may be accessed by multiple goroutines.

Reply:
Maybe you can use 2d array, get key array by keys[goroutine unique id]

Reply:
We can add an argument for the caller to pass in the key buffer.